tests/integration/src/test/scala/mesosphere/marathon/integration/MesosAppIntegrationTest.scala

```diff
@@ -12,14 +12,30 @@ import mesosphere.marathon.raml.{App, Container, DockerContainer, EngineType}
 import mesosphere.marathon.state.PathId._
 import mesosphere.marathon.state.{HostVolume, VolumeMount}
 import mesosphere.mesos.Constraints.hostnameField
-import mesosphere.{AkkaIntegrationTest, WhenEnvSet, WaitTestSupport}
+import mesosphere.{AkkaIntegrationTest, WaitTestSupport, WhenEnvSet}
+import org.scalatest.concurrent.PatienceConfiguration.Interval
 import play.api.libs.json.JsObject
 
 import scala.collection.immutable.Seq
 import scala.concurrent.duration._
 
 class MesosAppIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
 
+  override lazy val mesosNumMasters = 1
+  override lazy val mesosNumSlaves = 2
+
+  override def afterAll(): Unit = {
+    // We need to start all the agents for the teardown to be able to kill all the (UNREACHABLE) executors/tasks
+    mesosCluster.agents.foreach(_.start())
+    eventually { mesosCluster.state.value.agents.size shouldBe mesosCluster.agents.size }
+    super.afterAll()
+  }
+
+  override def beforeAll(): Unit = {
```

Comment:
maybe we can instead of beforeAll we can do before with
mesosCluster.agents(0).start()
mesosCluster.agents(1).stop()

it won't do anything if we are already in this state but it will centralize the cleaning logic to one place making sure all tests start in the same state. WDYT?

Reply:
Let's try this :)